qi-lib/flow/core/deforest-cps.rkt

```diff
@@ -0,0 +1,323 @@
+#lang racket/base
```

Comment:
This commit (0a79a3d (deforestation: split syntax matching and production,
2024-04-30)) confuses me: the subject talks about splitting syntax matching and
production, but the introduced changes don't seem like the refactoring I would
expect. Instead this seems to be the addition of the CPS deforestation method.
Can you clarify this commit message or purpose?